app/lib/katello/repo_discovery.rb

```diff
@@ -142,40 +143,38 @@ def docker_search
       @found.sort!
     end
 
-    def anemone_proxy_details
+    def spidr_proxy_details
       details = {}
 
       if proxy
-        details[:proxy_host] = proxy_host
-        details[:proxy_port] = proxy_port
-        details[:proxy_user] = proxy.username
-        details[:proxy_password] = proxy.password
+        details[:host] = proxy_host
+        details[:port] = proxy_port
+        details[:user] = proxy.username
+        details[:password] = proxy.password
       end
 
       details
     end
 
-    def http_crawl(resume_point)
-      resume_point_uri = URI(resume_point)
-      resume_point_uri.user = @upstream_username if @upstream_username
-      resume_point_uri.password = @upstream_password if @upstream_password
-
-      Anemone.crawl(resume_point_uri, anemone_proxy_details) do |anemone|
-        anemone.focus_crawl do |page|
-          @crawled << page.url.path
-
-          page.links.each do |link|
-            if link.path.ends_with?('/repodata/')
-              page_url = page.url.clone
-              page_url.user = nil
-              page_url.password = nil
-              @found << page_url.to_s
-            else
-              @to_follow << link.to_s if should_follow?(link.path)
-            end
-          end
-          page.discard_doc! #saves memory, doc not needed
-          []
+    def spidr_crawl_pages(url)
+      user, password = @upstream_username, @upstream_password
+      Spidr.site(url, proxy: spidr_proxy_details) do |spider|
+        spider.authorized.add(url, user, password) if user && password
+        spider.every_page do |page|
+          @crawled << page.url.to_s
+          process_page_urls(page.urls)
+          spider.skip_page!
+        end
+      end
+    end
+
+    def process_page_urls(urls)
+      urls.each do |url|
+        url = url.to_s
```

Comment:
I think this is a bad idea. The URL object is way more valuable and I'd only add it to `@to_follow` with `to_s`.

Reply:
We don't necessarily use the URL object properties once we have the list of URLs in the page. It's just matching strings after that point..Becomes easier to pass that around and work with it IMO?

Reply:
Updated it in latest push to pass URL object.